generators/server/templates/src/test/java/package/web/rest/WebTestConfiguration.java.ejs

```diff
@@ -0,0 +1,47 @@
+<%#
+Copyright 2013-2019 the original author or authors from the JHipster project.
+
+This file is part of the JHipster project, see https://www.jhipster.tech/
+for more information.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-%>
+package <%= packageName %>.web.rest;
+
+<%_ if (reactive) { _%>
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.test.web.reactive.server.WebTestClient;
+<%_ } else { _%>
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+<%_ } _%>
+
+@Configuration
+public class WebTestConfiguration {
+    <%_ if (reactive) { _%>
+    @Bean
+    public WebTestClient webTestClient(ApplicationContext context) {
+        return WebTestClient.bindToApplicationContext(context).build();
```

Comment:
I wonder if we should not try to use `@AutoConfigureWebTestClient` on each controller test instead (and `@AutoConfigureMockMvc` for non reactive application)

Reply:
I'll check out if that works!

Reply:
Looks good, I think we should go this way, I've adjusted the PR!